test/jdk/java/foreign/enablenativeaccess/TestEnableNativeAccess.java

```diff
@@ -0,0 +1,232 @@
+                { "panama_enable_native_access", PANAMA_MAIN, successNoWarning(), new String[]{"--enable-native-access=panama_module"} },
+                { "panama_enable_native_access_reflection", PANAMA_REFLECTION, successNoWarning(), new String[]{"--enable-native-access=panama_module"} },
+                { "panama_enable_native_access_invoke", PANAMA_INVOKE, successNoWarning(), new String[]{"--enable-native-access=panama_module"} },
+
+                { "panama_comma_separated_enable", PANAMA_MAIN, successNoWarning(), new String[]{"--enable-native-access=java.base,panama_module"} },
+                { "panama_comma_separated_enable_reflection", PANAMA_REFLECTION, successNoWarning(), new String[]{"--enable-native-access=java.base,panama_module"} },
+                { "panama_comma_separated_enable_invoke", PANAMA_INVOKE, successNoWarning(), new String[]{"--enable-native-access=java.base,panama_module"} },
+
+                { "panama_enable_native_access_warn", PANAMA_MAIN, successWithWarning(), new String[]{} },
+                { "panama_enable_native_access_warn_reflection", PANAMA_REFLECTION, successWithWarning(), new String[]{} },
+                { "panama_enable_native_access_warn_invoke", PANAMA_INVOKE, successWithWarning(), new String[]{} },
+
+                { "panama_no_unnamed_module_native_access", UNNAMED, successWithWarning(), new String[]{} },
+                { "panama_all_unnamed_module_native_access", UNNAMED, successNoWarning(), new String[]{"--enable-native-access=ALL-UNNAMED"} },
+        };
+    }
+
+    /**
+     * Checks an expected result with the output captured by the given
+     * OutputAnalyzer.
+     */
+    void checkResult(Result expectedResult, OutputAnalyzer outputAnalyzer) {
+        expectedResult.expectedOutput().forEach(outputAnalyzer::shouldContain);
+        expectedResult.notExpectedOutput().forEach(outputAnalyzer::shouldNotContain);
+        int exitValue = outputAnalyzer.getExitValue();
+        if (expectedResult.shouldSucceed()) {
+            assertTrue(exitValue == 0);
+        } else {
+            assertTrue(exitValue != 0);
+        }
+    }
+
+    /**
+     * Runs the test to execute the given test action. The VM is run with the
+     * given VM options and the output checked to see that it matches the
+     * expected result.
+     */
+    OutputAnalyzer run(String action, String cls, Result expectedResult, String... vmopts)
+            throws Exception
+    {
+        Stream<String> s1 = Stream.of(vmopts);
+        Stream<String> s2 = cls.equals(UNNAMED) ? Stream.of("--enable-preview", "-p", MODULE_PATH, cls, action)
+                : Stream.of("--enable-preview", "-p", MODULE_PATH, "-m", cls, action);
+        String[] opts = Stream.concat(s1, s2).toArray(String[]::new);
+        OutputAnalyzer outputAnalyzer = ProcessTools
+                .executeTestJava(opts)
+                .outputTo(System.out)
+                .errorTo(System.out);
+        if (expectedResult != null)
```

Comment:
I don't see how `expectedResult` could be `null` here? If it is `null` unexpectedly, I think it would be better to fail here (to avoid false-positives).

Reply:
Good point, I think it's safe to drop the null check as we don't pass null as `expectedResult` anywhere.